### PR TITLE
Fix body margin shifting with modals, fix error on project column edit

### LINKED ISSUE #30816: Show modal will cause body width becomeing smaller
### Description

image

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

build

### Database

None

### PR DESCRIPTION
**Pull request by silverwind**
Fixes: https://github.com/go-gitea/gitea/issues/30816, regression from https://github.com/go-gitea/gitea/pull/30723.
Fixes: https://github.com/go-gitea/gitea/pull/30815, regression from https://github.com/go-gitea/gitea/pull/30723.

Fomantic [expects a callback](https://github.com/fomantic/Fomantic-UI/blob/59d9b409879ad9413ea0a3efa4ab2e51017ad9b9/src/definitions/modules/modal.js#L530-L534) to be called during `hide` which we did not do, so it could never remove the margin it added to `body`.

I do observe the body content shifting to right by 1px when modal opens, but this is a bug that existed on v1.21 as well, so not a regression.

### CODE REVIEW

File: web_src/css/base.css [hunk 1]
```diff
@@ -67,6 +67,7 @@ body {
   flex-direction: column;
   overflow-x: visible;
   overflow-wrap: break-word;
+  scrollbar-gutter: stable;
```
Comment (silverwind) [inline web_src/css/base.css]:
```suggestion
```

File: web_src/js/modules/fomantic/dimmer.js [hunk 2]
```diff
@@ -3,8 +3,9 @@ import {queryElemChildren} from '../../utils/dom.js';
 
 export function initFomanticDimmer() {
   // stand-in for removed dimmer module
-  $.fn.dimmer = function (arg0, $el) {
+  $.fn.dimmer = function (arg0, arg1) {
     if (arg0 === 'add content') {
+      const $el = arg1;
       const existingDimmer = document.querySelector('body > .ui.dimmer');
       if (existingDimmer) {
         queryElemChildren(existingDimmer, '*', (el) => el.remove());
```
Comment (wxiaoguang) [inline web_src/js/modules/fomantic/dimmer.js]:
```suggestion
        queryElemChildren(existingDimmer, '*', (el) => el.classList.add('hidden'));
```

Then fix #30815

Comment (silverwind) [reply web_src/js/modules/fomantic/dimmer.js]: I checked behaviour on fomantic site, and yes, they do appear to just move modals, never remove them.

image

**Comment by silverwind** [comment]
> I do observe the body content shifting to right by 1px when modal opens, but this is a bug that existed on v1.21 as well, so not a regression.

Apparently this is only a problem in Firefox. No content shift in Chrome.

**Review by wxiaoguang — APPROVED** [review]

**Comment by silverwind** [comment]
> > I do observe the body content shifting to right by 1px when modal opens, but this is a bug that existed on v1.21 as well, so not a regression.
> 
> Apparently this is only a problem in Firefox. No content shift in Chrome. In Firefox there is a 1px difference in this calculation:
> image

Definitely a browser bug, not something we can fix. It does not happen at 100% page zoom level.

**Comment by silverwind** [comment]
> > I do observe the body content shifting to right by 1px when modal opens, but this is a bug that existed on v1.21 as well, so not a regression.
> 
> Apparently this is only a problem in Firefox. No content shift in Chrome. In Firefox there is a 1px difference in this calculation:
> image

I give up on this. Even Chrome shows a 1px diff in some zoom levels. Likely this is expected behaviour in browsers, idk.

image

**Comment by silverwind** [comment]
No backport as https://github.com/go-gitea/gitea/pull/30723 is not in v1.22.

**Review by lunny — APPROVED** [review]

**Pull request merged**